File: frontend/src/services/caseService.js
import api from './api';

class CaseService {
    async getAllCases() {
        try {
            const response = await api.get('/cases');
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch cases');
        }
    }

    async getCaseById(id) {
        try {
            const response = await api.get(`/cases/${id}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch case');
        }
    }

    async createCase(caseData) {
        try {
            const response = await api.post('/cases', caseData);
            return response.data;
        } catch (error) {
            throw new Error('Failed to create case');
        }
    }

    async updateCase(id, caseData) {
        try {
            const response = await api.put(`/cases/${id}`, caseData);
            return response.data;
        } catch (error) {
            throw new Error('Failed to update case');
        }
    }

    async deleteCase(id) {
        try {
            await api.delete(`/cases/${id}`);
        } catch (error) {
            throw new Error('Failed to delete case');
        }
    }

    async getCasesByStatus(status) {
        try {
            const response = await api.get(`/cases/status/${status}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch cases by status');
        }
    }

    async getCasesByAssignedTo(assignedToId) {
        try {
            const response = await api.get(`/cases/assigned/${assignedToId}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch cases by assigned user');
        }
    }

    async closeCase(id) {
        try {
            const response = await api.put(`/cases/${id}/close`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to close case');
        }
    }
}

export default new CaseService(); 

File: backend/src/main/java/com/crms/case_management/CaseService.java
package com.crms.case_management;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
public class CaseService {

    @Autowired
    private CaseRepository caseRepository;

    public List<Case> getAllCases() {
        return caseRepository.findAll();
    }

    public Optional<Case> getCaseById(Long id) {
        return caseRepository.findById(id);
    }

    public Case createCase(Case caseEntity) {
        return caseRepository.save(caseEntity);
    }

    public Case updateCase(Long id, Case caseDetails) {
        Case caseEntity = caseRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Case not found"));

        caseEntity.setAssignedTo(caseDetails.getAssignedTo());
        caseEntity.setStatus(caseDetails.getStatus());
        caseEntity.setNotes(caseDetails.getNotes());

        if ("Closed".equals(caseDetails.getStatus())) {
            caseEntity.setClosedAt(LocalDateTime.now());
        }

        return caseRepository.save(caseEntity);
    }

    public void deleteCase(Long id) {
        caseRepository.deleteById(id);
    }

    public List<Case> getCasesByStatus(String status) {
        return caseRepository.findByStatus(status);
    }

    public List<Case> getCasesByAssignedTo(Long assignedToId) {
        return caseRepository.findByAssignedToId(assignedToId);
    }

    public Case closeCase(Long id) {
        Case caseEntity = caseRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Case not found"));
        
        caseEntity.setStatus("Closed");
        caseEntity.setClosedAt(LocalDateTime.now());
        
        return caseRepository.save(caseEntity);
    }
} 